packages/rich-text/src/plugins/shared/FetchingWrappedAssetCard.tsx

```diff
@@ -30,8 +30,24 @@ interface FetchingWrappedAssetCardProps {
   sdk: FieldExtensionSDK;
 }
 
-function downloadAsset(url: string) {
-  window.open(url, '_blank', 'noopener,noreferrer');
+function downloadAsset(url: string, fileName: string) {
+  fetch(url, {
+    method: 'GET',
+    headers: {},
+  })
+    .then((response) => {
+      response.arrayBuffer().then(function (buffer) {
+        const url = window.URL.createObjectURL(new Blob([buffer]));
+        const link = document.createElement('a');
+        link.href = url;
+        link.setAttribute('download', fileName); //or any other extension
+        document.body.appendChild(link);
+        link.click();
+      });
+    })
+    .catch((err) => {
+      console.log(err);
```

Comment:
Maybe show a Froam36 notification like "Failed to download the asset" AND re-throw the error so the app can catch it and report to Sentry. Worth checking if throwing an error crashes only the asset card or the whole app.